src/api/routes/core-node-rpc-proxy.ts

```diff
@@ -159,17 +177,89 @@ export const CoreNodeRpcProxyRouter: FastifyPluginAsync<
     }
   );
 
-  let feeEstimationModifier: number | null = null;
-  fastify.addHook('onReady', () => {
-    const feeEstEnvVar = process.env['STACKS_CORE_FEE_ESTIMATION_MODIFIER'];
-    if (feeEstEnvVar) {
-      const parsed = parseFloat(feeEstEnvVar);
-      if (!isNaN(parsed) && parsed > 0) {
-        feeEstimationModifier = parsed;
-      }
+  // Fee estimator options
+  const feeOpts: FeeEstimateProxyOptions = {
+    estimationModifier: DEFAULT_FEE_ESTIMATION_MODIFIER,
+    pastTenureFullnessWindow: DEFAULT_FEE_PAST_TENURE_FULLNESS_WINDOW,
+    pastDimensionFullnessThreshold: DEFAULT_FEE_PAST_DIMENSION_FULLNESS_THRESHOLD,
+    currentDimensionFullnessThreshold: DEFAULT_FEE_CURRENT_DIMENSION_FULLNESS_THRESHOLD,
+    currentBlockCountMinimum: DEFAULT_FEE_CURRENT_BLOCK_COUNT_MINIMUM,
+    readCountLimit: DEFAULT_BLOCK_LIMIT_READ_COUNT,
+    readLengthLimit: DEFAULT_BLOCK_LIMIT_READ_LENGTH,
+    writeCountLimit: DEFAULT_BLOCK_LIMIT_WRITE_COUNT,
+    writeLengthLimit: DEFAULT_BLOCK_LIMIT_WRITE_LENGTH,
+    runtimeLimit: DEFAULT_BLOCK_LIMIT_RUNTIME,
+    sizeLimit: BLOCK_LIMIT_SIZE,
+    minTxFeeRatePerByte: MINIMUM_TX_FEE_RATE_PER_BYTE,
+  };
+  fastify.addHook('onReady', async () => {
+    feeOpts.estimationModifier =
+      parseFloatEnv('STACKS_CORE_FEE_ESTIMATION_MODIFIER') ?? feeOpts.estimationModifier;
+    feeOpts.pastTenureFullnessWindow =
+      parseFloatEnv('STACKS_CORE_FEE_PAST_TENURE_FULLNESS_WINDOW') ??
+      feeOpts.pastTenureFullnessWindow;
+    feeOpts.pastDimensionFullnessThreshold =
+      parseFloatEnv('STACKS_CORE_FEE_PAST_DIMENSION_FULLNESS_THRESHOLD') ??
+      feeOpts.pastDimensionFullnessThreshold;
+    feeOpts.currentDimensionFullnessThreshold =
+      parseFloatEnv('STACKS_CORE_FEE_CURRENT_DIMENSION_FULLNESS_THRESHOLD') ??
+      feeOpts.currentDimensionFullnessThreshold;
+    feeOpts.currentBlockCountMinimum =
+      parseFloatEnv('STACKS_CORE_FEE_CURRENT_BLOCK_COUNT_MINIMUM') ??
+      feeOpts.currentBlockCountMinimum;
+
+    // Get block limits from current Stacks epoch PoX info.
+    const client = new StacksCoreRpcClient();
+    const poxData = await client.getPox();
```

Comment:
This RPC request is pretty likely to fail in many situations. We've ran into a lot of deadlock type scenarios in the past when we've expected the stacks-node RPC to be ready when the API launches.

I'd recommend firing this off in the background, and include a try/catch + retry mechanism. Probably also some info level logs indicating the configured RPC endpoint that it's trying to reach and connection error details.

Reply:
I'd also maybe consider moving this into a lazy eval that only happens once the data is needed, e.g. on the first `POST /v2/transaction` proxy intercept to modify fees.

Reply:
> I'd recommend firing this off in the background, and include a try/catch + retry mechanism

Good idea, will do

> I'd also maybe consider moving this into a lazy eval that only happens once the data is needed

That sounds good to me too, however the first call to that endpoint will have the default values since it will fire off the pox call async. I guess that's not a big deal, right?